src/runtime/client.ts

```diff
@@ -51,16 +51,13 @@ export async function useAsyncQuery<
 ): Promise<AsyncData<PickFrom<ReturnType<Transform>, PickKeys>, TError>> {
   const { $client } = useNuxtApp()
   const key = getQueryKey(pathAndInput)
-  const serverError = useState<TError | null>(`error-${key}`, () => null)
   const { error, data, ...rest } = await useAsyncData(
     key,
     () => $client.query(...pathAndInput),
     // @ts-expect-error: Internal
     options,
-  )
-
-  if (process.server && error.value && !serverError.value)
-    serverError.value = error.value as any
+    )
+  const serverError = useState<TError | null>(`error-${key}`, () => error.value)
```

Comment:
Getting `nuxt instance unavailable` when trying this locally.

Probably related to https://github.com/nuxt/framework/issues/6996

Reply:
Guess that's related to SSR.  
I reverted the changes and  removed `process.server` on line 62 to achieve the same effect. The error in playground should be gone now.